pkg/pipelines/initialise.go

```diff
@@ -0,0 +1,247 @@
+		return err
+	}
+
+	if err := addKustomize("bases", []string{}, filepath.Join(gitopsPath, envsDir, baseDir, kustomize), o.Prefix); err != nil {
+		return err
+	}
+
+	fmt.Println("\nGitops initialised")
+	return nil
+}
+
+func getCICDDir(path, prefix string) string {
+	return filepath.Join(path, envsDir, addPrefix(prefix, cicdDir))
+}
+
+func createPipelineResources(namespaces map[string]string, gitopsRepo, prefix string) map[string][]interface{} {
+
+	// key: path of the resource
+	// value: YAML content of the resource
+	outputs := make(map[string][]interface{}, 0)
+
+	// create namespace
+	outputs[namespacesPath] = append(outputs[namespacesPath], createNamespace(namespaces["cicd"]))
+
+	// create roles and rolebindings for pipeline service account
+	role := createClusterRole(meta.NamespacedName("", clusterRoleName), rules)
+	outputs[rolesPath] = append(outputs[rolesPath], role)
+
+	sa := createServiceAccount(meta.NamespacedName(namespaces["cicd"], saName))
+	rolebinding := createRoleBinding(meta.NamespacedName(namespaces["cicd"], roleBindingName), sa, role.Kind, role.Name)
+	outputs[rolebindingsPath] = append(outputs[rolebindingsPath], rolebinding)
+
+	// create deploy from source task
+	task := tasks.GenerateDeployFromSourceTask(namespaces["cicd"], getPipelinesDir("", prefix))
+	outputs[tasksPath] = append(outputs[tasksPath], task)
+
+	// create ci-cd pipelines for gitops repo
+	ciPipeline := createStageCIPipeline(meta.NamespacedName(namespaces["cicd"], "ci-dryrun-from-pr-pipeline"), namespaces["cicd"])
+	outputs[ciPipelinesPath] = append(outputs[ciPipelinesPath], ciPipeline)
+
+	cdPipeline := createStageCDPipeline(meta.NamespacedName(namespaces["cicd"], "cd-deploy-from-push-pipeline"), namespaces["cicd"])
+	outputs[cdPipelinesPath] = append(outputs[cdPipelinesPath], cdPipeline)
+
+	// create trigger bindings
+	prBinding := triggers.CreatePRBinding(namespaces["cicd"])
+	outputs[prBindingPath] = append(outputs[prBindingPath], prBinding)
+
+	pushBinding := triggers.CreatePushBinding(namespaces["cicd"])
+	outputs[pushBindingPath] = append(outputs[pushBindingPath], pushBinding)
+
+	// create trigger templates
+	prTemplate := triggers.CreateStageCIDryRunTemplate(namespaces["cicd"], saName)
+	outputs[prTemplatePath] = append(outputs[prTemplatePath], prTemplate)
+
+	pushTemplate := triggers.CreateStageCDPushTemplate(namespaces["cicd"], saName)
+	outputs[pushTemplatePath] = append(outputs[pushTemplatePath], pushTemplate)
+
+	// create eventlisteners and route
+	eventListener := eventlisteners.Generate(gitopsRepo, namespaces["cicd"], saName)
+	outputs[eventListenerPath] = append(outputs[eventListenerPath], eventListener)
+
+	route := routes.Generate(namespaces["cicd"])
+	outputs[routePath] = append(outputs[routePath], route)
+
+	return outputs
+}
+
+func writeResources(path string, prefix string, files map[string][]interface{}) ([]string, error) {
```

Comment:
I don't think you actually need to add the prefix to all the files.

The resources need to be prefixed, but not necessarily the filenames.